From 2f3368f46bcde72c5062f5586209032a6508f292 Mon Sep 17 00:00:00 2001
From: Artem Bityutskiy <artem.bityutskiy@linux.intel.com>
Date: Tue, 26 Jul 2022 18:29:35 +0300
Subject: [PATCH 12/18] tools/power turbostat: do not decode ACC for ICX and
 SPR

commit 6287e6f0fdd36be4bbde6f539df6ea85eb2476c2 upstream.

The ACC (automatic C-state conversion) feature was available on Sky Lake and
Cascade Lake Xeons (SKX and CLX), but it is not available on Ice Lake and
Sapphire Rapids Xeons (ICX and SPR). Therefore, stop decoding it for ICX and
SPR.

Signed-off-by: Artem Bityutskiy <artem.bityutskiy@linux.intel.com>
Signed-off-by: Len Brown <len.brown@intel.com>
[Ankit: Backported to v4.19.x]
Upstream commit basically removes icelake_x and sapphirerapids_x support from
automatic_cstate_conversion_probe(). is_icx() and is_spr() has the same macro
check.

int is_icx(unsigned int family, unsigned int model)
{

	if (!genuine_intel)
		return 0;

	if (family != 6)
		return 0;

	switch (model) {
	case INTEL_FAM6_ICELAKE_X:
		return 1;
	}
	return 0;
}

int is_spr(unsigned int family, unsigned int model)
{

	if (!genuine_intel)
		return 0;

	if (family != 6)
		return 0;

	switch (model) {
	case INTEL_FAM6_SAPPHIRERAPIDS_X:
		return 1;
	}
	return 0;
}

So removing these function call effectively does the same thing on v4.19 kernel version.
Signed-off-by: Ankit Jain <ankitja@vmware.com>
---
 tools/power/x86/turbostat/turbostat.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tools/power/x86/turbostat/turbostat.c b/tools/power/x86/turbostat/turbostat.c
index 0bf1de6fd081..90edbc1765c2 100644
--- a/tools/power/x86/turbostat/turbostat.c
+++ b/tools/power/x86/turbostat/turbostat.c
@@ -4014,8 +4014,7 @@ void perf_limit_reasons_probe(unsigned int family, unsigned int model)
 
 void automatic_cstate_conversion_probe(unsigned int family, unsigned int model)
 {
-	if (is_skx(family, model) || is_bdx(family, model) ||
-	    is_icx(family, model) || is_spr(family, model))
+	if (is_skx(family, model) || is_bdx(family, model))
 		has_automatic_cstate_conversion = 1;
 }
 
-- 
2.23.1

